addons/auth_totp_mail_enforce/controllers/home.py

```diff
@@ -0,0 +1,28 @@
+from datetime import datetime, timedelta
+
+import odoo.addons.auth_totp.controllers.home
+
+from odoo import http
+from odoo.exceptions import AccessDenied
+from odoo.http import request
+
+
+class Home(odoo.addons.auth_totp.controllers.home.Home):
+    @http.route()
+    def web_totp(self, redirect=None, **kwargs):
+        response = super().web_totp(redirect=redirect, **kwargs)
+        if not response.is_qweb or response.qcontext['user'].totp_enabled:
+            return response
+
+        # If no authentication code sent by email in last hour, send email
+        # If user demand, send email
+        if not request.env['auth.totp.rate.limit.log'].sudo().search_count([
+            ('user_id', '=', response.qcontext['user'].id),
+            ('create_date', '>=', datetime.now() - timedelta(seconds=3600)),
```

Comment:
didn't you mean to filter on `send_email` types?
```suggestion
        if not request.env['auth.totp.rate.limit.log'].sudo().search_count([
            ('user_id', '=', response.qcontext['user'].id),
            ('limit_type', '=', 'send_email'),
            ('create_date', '>=', datetime.now() - timedelta(seconds=3600)),
```

The comments could also be slightly clearer about the fact that the "no email in last hour" test is used to trigger the initial email without causing a new email at every refresh. 

I'm not 100% sure we need this, maybe it's just acceptable that the email is sent again if the user ever refreshes the page. After all, they have proven that they know the password, so we're pretty sure this is legit. And if it's not, well the possibility of spam is the least of the user's problems.

PS: if we keep the search, the code may be more readable by introducing an intermediary `mfa_email_count` variable to be used in the `if` condition.